### ISSUE TITLE
An error occurs if the orderBy array of findMany() contains an empty object

### ISSUE THREAD

**Issue by kensukesaito** [prisma/prisma#5215]
## Bug description

Following error occurs if the orderBy array of findMany() contains an empty object

```
Error: 
Invalid `prisma.example.findMany()` invocation:

{
  orderBy: [
    {
      createdAt: undefined
    }
  ]
}

Unknown arg `0` in orderBy.0 for type ExampleOrderByInput. Available args:

type ExampleOrderByInput {
  id?: SortOrder
  createdAt?: SortOrder
}
```

## How to reproduce

```typescript
// ❌ Error
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: args.orderBy?.createdAt?.direction // 'asc' | 'desc' | undefined
    }
  ]
})

// ❌ Error
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: undefined
    }
  ]
})

// ❌ Error
const examples = await prisma.example.findMany({
  orderBy: [
    {
    }
  ]
})

// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: {
    createdAt: undefined
  }
})

// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: [
  ]
})

// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: args.orderBy?.createdAt?.direction // 'asc' | 'desc' | undefined
    }
  ].filter((value) => !Object.values(value).includes(undefined))
})
```

## Expected behavior

```typescript
// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: undefined
    }
  ]
})
```

## Prisma information

```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Example {
  /// The value of this field is generated by the database as: `uuid_generate_v4()`.
  id                                  String    @id @default(dbgenerated())
  createdAt                           DateTime  @default(now())
}
```

## Environment & setup

- OS: Amazon Linux on AWS Lambda
- Database: PostgreSQL on Amazon RDS
- Node.js version: 12.x
- Prisma version: 2.15.0

**Comment by pantharshit00** [comment]
I will say this is expected but we are not throwing an error on a type system level here. I think that is what we should fix. The runtime error is correct. Any field which you pass as undefined will be omitted(please see: https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined)

I will mark this as confirmed bug for now and we can see how we can improve the types here.

**Comment by kensukesaito** [comment]
@pantharshit00 
> I will mark this as confirmed bug for now and we can see how we can improve the types here.

Thanks.

> I will say this is expected
> Any field which you pass as undefined will be omitted

The following code works.

```typescript
// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: {
    createdAt: undefined
  }
})
```

However, the following does not.

```typescript
// ❌ Error
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: undefined
    }
  ]
})
```

It would be really helpful to get the latter code working as well.

I don't want to add tricky code like the following to get around this.

```typescript
// ✅ Works
const examples = await prisma.example.findMany({
  orderBy: [
    {
      createdAt: args.orderBy?.createdAt?.direction // 'asc' | 'desc' | undefined
    }
  ].filter((value) => !Object.values(value).includes(undefined))
})
```

**Comment by pantharshit00** [comment]
This is still reproducible in latest version.

**Comment by matthewmueller** [comment]
_(Unassigning myself because we originally thought it was expected and I'd write a comment, but [@pantharshit00 clarified that point earlier](https://github.com/prisma/prisma/issues/5215#issuecomment-769894011))_

@kennhung curious why you're running into this in your application? Perhaps a filter pattern where `created_at` can be provided or undefined, depending on what the user chooses?

**Comment by kennhung** [comment]
@matthewmueller
I think you have mentioned the wrong person?

**Comment by matthewmueller** [comment]
Sorry about that! @kensukesaito curious why you're running into this in your application? Perhaps a filter pattern where created_at can be provided or undefined, depending on what the user chooses?

**Comment by kensukesaito** [comment]
Thanks, @matthewmueller 

> Perhaps a filter pattern where created_at can be provided or undefined, depending on what the user chooses?

Yes.

**Comment by kensukesaito** [comment]
I confirmed this issue has been fixed on v5.7.1. Thanks.
